apps/paymaster-proxy/src/e2eTests/uniquePolicy_sponsorUserOperation.spec.ts

```diff
@@ -0,0 +1,384 @@
+import crypto from 'crypto'
+import {
+  deepHexlify,
+  ENTRYPOINT_ADDRESS_V06,
+  ENTRYPOINT_ADDRESS_V07,
+} from 'permissionless/utils'
+import supertest from 'supertest'
+import { parseGwei, toHex } from 'viem'
+import { baseSepolia, optimismSepolia, sepolia, zoraSepolia } from 'viem/chains'
+import { afterAll, beforeAll, describe, expect, it } from 'vitest'
+
+import { fraxtalSepolia } from '@/constants/fraxtalSepolia'
+import {
+  E2E_TEST_ADMIN_BASE_URL,
+  E2E_TEST_BASE_URL,
+} from '@/e2eTests/e2eTestBaseUrl'
+import {
+  getMintUserOperationWithRandomSimpleAccount,
+  getRevertingUserOperationWithRandomSimpleAccount,
+} from '@/testUtils/simpleAccount/exampleSimpleAccountUserOperations'
+import { successfulPaymasterResultMatcher } from '@/testUtils/successfulPaymasterResultMatcher'
+
+const app = supertest(E2E_TEST_BASE_URL)
+
+const adminApp = supertest(E2E_TEST_ADMIN_BASE_URL)
+
+describe('pm_sponsorUserOperation', async () => {
+  describe.each([
+    zoraSepolia,
+    optimismSepolia,
+    baseSepolia,
+    fraxtalSepolia,
+    sepolia,
+  ])('$name', async (chain) => {
+    let apiKey: string
+
+    beforeAll(async () => {
+      const entityId = crypto.randomUUID()
+
+      const registerApiKeyResult = await adminApp
+        .post('/admin/createPaymasterApiKey')
+        .send({
+          chainId: chain.id,
+          entityId,
+        })
+        .set('Content-Type', 'application/json')
+        .set('Accept', 'application/json')
+      apiKey = registerApiKeyResult.body.apiKey
+    })
```

Comment:
creates a real apiKey to be used for the duration of the test, then deletes it at the end of the test